src/lib/one-graph/netlify-graph-code-exporter-snippets.js

```diff
@@ -0,0 +1,543 @@
+      })
+      .filter(Boolean)
+
+    const nestedNamedFragments = selections.reduce((acc, selection) => {
+      if (
+        (selection.kind === 'Field' || selection.kind === 'SelectionNode' || selection.kind === 'InlineFragment') &&
+        selection.selectionSet !== undefined
+      ) {
+        return [...acc, ...findReferencedFragments(selection.selectionSet)]
+      }
+      return acc
+    }, [])
+
+    return [...namedFragments, ...nestedNamedFragments]
+  }
+
+  const { selectionSet } = definition
+
+  return findReferencedFragments(selectionSet)
+}
+
+const operationDataByName = (graph, name) => graph.find((operationData) => operationData.name === name)
+
+const topologicalSortHelper = ({ graph, node, temp, visited }, result) => {
+  temp[node.name] = true
+  const neighbors = node.fragmentDependencies
+  neighbors.forEach((fragmentDependency) => {
+    const fragmentOperationData = operationDataByName(graph, fragmentDependency.name.value)
+
+    if (!fragmentOperationData) {
+      return
+    }
+
+    if (temp[fragmentOperationData.name]) {
+      console.error('The operation graph has a cycle')
+      return
+    }
+    if (!visited[fragmentOperationData.name]) {
+      topologicalSortHelper(
+        {
+          node: fragmentOperationData,
+          visited,
+          temp,
+          graph,
+        },
+        result,
+      )
+    }
+  })
+  temp[node.name] = false
+  visited[node.name] = true
+  result.push(node)
+}
+
+const toposort = (graph) => {
+  const result = []
+  const visited = {}
+  const temp = {}
+  graph.forEach((node) => {
+    if (!visited[node.name] && !temp[node.name]) {
+      topologicalSortHelper({ node, visited, temp, graph }, result)
+    }
+  })
+  return result
+}
+
+const computeOperationDataList = ({ query, variables }) => {
+  const operationDefinitions = getOperationNodes(query)
+
+  const fragmentDefinitions = []
+
+  operationDefinitions.forEach((operationDefinition) => {
+    if (operationDefinition.kind === 'FragmentDefinition') {
+      fragmentDefinitions.push(operationDefinition)
+    }
+  })
+
+  const rawOperationDataList = operationDefinitions.map((operationDefinition) => ({
+    query: print(operationDefinition),
+    name: getOperationName(operationDefinition),
+    displayName: getOperationDisplayName(operationDefinition),
+    // $FlowFixMe: Come back for this
+    type: operationDefinition.operation || 'fragment',
+    variableName: formatVariableName(getOperationName(operationDefinition)),
+    variables: getUsedVariables(variables, operationDefinition),
+    operationDefinition,
+    fragmentDependencies: findFragmentDependencies(fragmentDefinitions, operationDefinition),
+  }))
+
+  const operationDataList = toposort(rawOperationDataList)
+
+  return {
+    operationDefinitions,
+    fragmentDefinitions,
+    rawOperationDataList,
+    operationDataList,
+  }
+}
+
+const capitalizeFirstLetter = (string) => string.charAt(0).toUpperCase() + string.slice(1)
```

Comment:
As you're using these in a few places it would be nice to move the helpers into a separate file

Reply:
This file should remain self-contained since it's going away shortly.